futures-util/src/future/abortable.rs

```diff
@@ -0,0 +1,131 @@
+use futures_core::future::Future;
+use futures_core::task::{self, AtomicWaker, Poll};
+use std::marker::Unpin;
+use std::mem::PinMut;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicBool, Ordering};
+
+/// A future which can be remotely short-circuited using an `AbortHandle`.
+#[derive(Debug, Clone)]
+#[must_use = "futures do nothing unless polled"]
+pub struct Abortable<T> {
+    future: T,
```

Comment:
Instead of `T`, consider using `Fut` as type param. I'm currently introducing this convention in my PRs. I've found it to be really helpful when reading the code later.
`F` function, `Fut` future, `St` stream, `Si` sink, `T` output, `R` `AsyncRead`, `W` `AsyncWrite`, `E` error, `S` sink + stream

Reply:
Done.